stargate/pkg/auth/client.go

```diff
@@ -0,0 +1,96 @@
+package auth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type TableBasedTokenProvider struct {
+	client   *client
+	username string
+	password string
+}
+
+type client struct {
+	serviceURL string
+	httpClient *http.Client
+}
+
+type AuthResponse struct {
+	AuthToken string `json:"authToken"`
+}
+
+type AuthRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+func NewTableBasedTokenProvider(serviceURL, username, password string) TableBasedTokenProvider {
```

Comment:
`TableBasedTokenProvider` doesn't need to be public. It's implementation can be private then `NewTableBasedTokenProvider()` would look like:

```
func NewTableBasedTokenProvider(serviceURL, username, password string)  AuthProviderIFace {

```

Reply:
done